test/cp.js

```diff
@@ -451,6 +451,45 @@ test('-R implies -P', t => {
   });
 });
 
+test('-Ru respects the -u flag recursively (don\'t update newer file)', t => {
+  const dir = `${t.context.tmp}/cp-Ru`;
+  const sourceDir = `${dir}/old`;
+  const sourceFile = `${sourceDir}/file`;
+  const destDir = `${dir}/new`;
+  const destFile = `${destDir}/file`;
+  [sourceDir, destDir].forEach(d => shell.mkdir('-p', d));
+  shell.ShellString('foo\n').to(sourceFile);
+  shell.ShellString('bar\n').to(destFile);
```

Comment:
I would like to separate setup from test logic. Two (alternative) suggestions:

 1. Instead of setting up at runtime, commit this directory structure under `test/resources/`, or
 2. Add an empty line between setup code and test logic (possibly add a comment above such as `// Setup code`)

---

You'll need to continue modifying the time stamps at runtime, as you have here.